training/tf/parse.py

```diff
@@ -30,203 +30,90 @@
 import time
 import tensorflow as tf
 from tfprocess import TFProcess
+from chunkparser import ChunkParser
 
-DATA_ITEM_LINES = 121
-SKIP = 16
-CHUNK_PASSES = 2
+SKIP = 8
+RAM_BATCH_SIZE = 2048
```

Comment:
Does this really fit in the GPU RAM!?

Note that the memory for the batch itself is ~nothing: The common problem is that TensorFlow by default accumulates all the gradients from each example separately and then sums them up at the end.

So this effectively results in 2048 copies of the network in GPU RAM, while probably only works for either very small networks or uncommonly huge hardware.

Reply:
Oh darn. You're right that makes no sense. I've got 2 1080Ti, but this uses 1. What would be a sane value?

Reply:
The current net has 4'214'045 weights, so about 16M. I guess RAM_BATCH_SIZE=512 should work?

Reply:
It seems a RAM_BATCH_SIZE of 512 results in only 1.3G gpu usage with tensorflow? Do you know how that's possible?